code/modules/bluespace_exploration/encounter/ship_datum.dm

```diff
@@ -0,0 +1,139 @@
+			var/turf/open/floor/F = T
+			calculated_health += 1
+			if(!F.broken && !F.burnt)
+				calculated_health += 1
+			if(!istype(T, /turf/open/floor/plating))
+				calculated_health += 2
+	if(calculated_health > max_ship_integrity)
+		max_ship_integrity = calculated_health
+	current_ship_integrity = calculated_health
+	var/difference = max_ship_integrity - current_ship_integrity
+	var/used_health_pool = max_ship_integrity * health_percentage
+	integrity_remaining = used_health_pool - difference
+
+/datum/ship_datum/proc/hull_critical(list/turfs)
+	if(critical)
+		return
+	critical = TRUE
+	if(!LAZYLEN(turfs))
+		WARNING("Ship [ship_name], port_id [mobile_port_id] has no turfs.")
+		return
+	var/turf/first_turf = turfs[1]
+	message_admins("The [ship_name], port_id [mobile_port_id] has been destroyed, at [ADMIN_JMP(first_turf)]")
+	log_attack("The [ship_name], port_id [mobile_port_id] has been destroyed!")
+	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(mobile_port_id, FALSE)
+	if(M)
+		WARNING("Warning, no docking port located on shuttle [ship_name]")
+		qdel(M, force = TRUE)
+	for(var/turf/T in turfs)
+		for(var/obj/machinery/light/L in T)
+			L.force_emergency_mode = TRUE
+			L.update()
+		//Play an alarm to anyone / any observers on the shuttle
+		for(var/mob/smob in T)
+			if(smob.client)
+				SEND_SOUND(smob.client, 'sound/machines/alarm.ogg')
+	addtimer(CALLBACK(src, .proc/destroy_ship, turfs), 140)
+
+/datum/ship_datum/proc/destroy_ship(list/turfs)
+	set waitfor = FALSE
+	var/exploded = FALSE
+	for(var/turf/T in turfs)
+		for(var/obj/machinery/bluespace_drive/BS in T)
+			if(!exploded)
+				//Blow up the bluespace drive (basically the ships reactor / core)
+				explosion(T, 12, 15, 18, -1, FALSE)
+				exploded = TRUE
+			qdel(BS)
+		for(var/obj/machinery/power/apc/A in T)
+			//No more power
+			A.set_broken()
+	//No explosion, explode anyway
+	if(!exploded)
+		explosion(pick(turfs), 12, 15, 18, -1, FALSE)
+	if(mobile_port_id == "exploration")
+		print_command_report("Sensors of your station's exploration shuttle, The Pathfinder have gone dark. The ship and its crew are assumed lost. Their bodies could potentially be recovered, however their last known sector is known to be dangerous.")
+	qdel(src)
+
+//Removes all the star systems we could jump to, and generates new ones
+//Called: on ship jump
+/datum/ship_datum/proc/recalculate_star_systems()
+	//Don't initialize until we need it, byond memory doesn't come cheap
+	if(!islist(star_systems))
+		star_systems = list()
+	else
+		star_systems.Cut()
+	for(var/i = 0 to 5)
+		//Generate star systems
+		var/datum/star_system/system = new(jumps)
+		system.bluespace_ruins = bluespace
+		//Note: If another system has the same name, it will be overwritten which is fine.
+		star_systems[system.name] = system
+	jumps ++
+
+/datum/ship_datum/custom_shuttle
+	bluespace = FALSE
+	//Custom shuttles have to take a really, really heavy beating to be considered destroyed
+	health_percentage = 0.8
```

Comment:
Seems like this incentivizes building a big empty box.

Reply:
It doesn't really matter since such a large amount has to get destroyed that a small shuttle or a huge shuttle will almost never be considered destroyed.
Plus that's a good thing imo, since building a large empty box requires building more thrusters, and is a lot more work than just RCDing more floor